pkg/node/provisioners/drive_provisioner.go

```diff
@@ -224,7 +224,15 @@ func (d *DriveProvisioner) GetVolumePath(vol *api.Volume) (string, error) {
 	}
 	volumeUUID, _ = util.GetVolumeUUID(volumeUUID)
 
-	partNum, err := d.partOps.SearchPartName(device, volumeUUID)
+	// For volumes that are in Published status, ignore partition re-read and
+	// get the partition number directly.
+	var partNum string
+	if vol.CSIStatus == apiV1.Published {
+		partNum, err = d.partOps.GetPartitionNameByUUID(device, volumeUUID)
```

Comment:
Please don't ignore errors here, it is not clear which call the error handling below belongs to

Reply:
I assume if/else block can combine to one error handling,  the purpose is to get partition name.  so the error handleing below belongs to the above if/else block. 
do you mean we should logging  2 different error logs in if/else ?

Reply:
one call one error handling

Reply:
how about below logic,  is it ok ? 
```
229     var handleFunc func(string, string) (string, error)  
230     if vol.CSIStatus == apiV1.Published {  
231         handleFunc = d.partOps.GetPartitionNameByUUID  
232     } else {  
233         handleFunc = d.partOps.SearchPartName                        
234     }     
235           
236     partNum, err := handleFunc(device, volumeUUID)   
237                                                      
238     if err != nil {  
239         // on device disconnect or node reboot device name might change and we need to re-sync drive info   
240         return "", fmt.Errorf("unable to find part name for device %s by uuid %s: %w", device, volumeUUID, err)  
241     }             
```

Reply:
Checkout example below: 
```golang
var partNum string
if partNum, err = d.partOps.SearchPartName(device, volumeUUID); err != nil {
  return "", fmt.Errorf("search part name for %s by uuid %s: %w", device, volumeUUID, err)
}
if vol.CSIStatus == apiV1.Published {
  if partNum, err = d.partOps.GetPartitionNameByUUID(device, volumeUUID); err != nil {
    return "", fmt.Errorf("get part name by uuid %s for device:%s : %w", volumeUUID, device, err)
  } 
}
```

Reply:
thank you @rkiselenko . changed the code, please help review again.

Reply:
Hi @rkiselenko,  changed the code as your suggestions,  could you please help review ?